### PR TITLE
Add new package: micromamba

### PR DESCRIPTION
**Pull request by charmoniumQ**
Micromamba is a C++ rewrite of the conda package manager that aims to be faster and more hermetic. Miniconda is written in Python, so it's installation process runs a script that downloads its own Python and Python packages (neither of which are managed by Spack). Micromamba is a C++ standalone rewrite of Miniconda that does not need to install Python or Python packages.

### CODE REVIEW

**Comment by charmoniumQ** [comment]
Currently blocked by https://github.com/spack/spack/issues/34193, although I am looking for workarounds.

**Comment by charmoniumQ** [comment]
Getting this error:

```
cd /tmp/sam/spack-stage/spack-stage-micromamba-1.0.0-hsyfletrgwe3rpkmoisedxw2zts5r7it/spack-build-hsyflet/libmamba && /home/sam/.local/share/spack/lib/spack/env/gcc/g++ -DLIBMAMBA_EXPORTS -DSPDLOG_FMT_EXTERNAL -Dlibmamba_EXPORTS -I/tmp/sam/spack-stage/spack-stage-micromamba-1.0.0-hsyfletrgwe3rpkmoisedxw2zts5r7it/spack-src/libmamba/include -I/tmp/sam/spack-stage/spack-stage-micromamba-1.0.0-hsyfletrgwe3rpkmoisedxw2zts5r7it/spack-src/libmamba/src -isystem /home/sam/.local/share/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/yaml-cpp-0.7.0-zm5yle4qzcg77uulpxar7fapzmmf2ns6/include -isystem /home/sam/.local/share/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/libreproc-14.2.4-lbugs4waibjhtzoaqh5rxsvn5lqutyjw/include -isystem /home/sam/.local/share/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/tl-expected-master-a2awf2nwoihkbpron67iobxi76gzdgme/include -isystem /home/sam/.local/share/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/tl-expected-master-a2awf2nwoihkbpron67iobxi76gzdgme/share -isystem /home/sam/.local/share/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/fmt-9.1.0-xki4etfiiw7rmyi64afr6tjxcso3iwvp/include -isystem /home/sam/.local/share/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/spdlog-1.10.0-k7cacussi2trtrd3h3v7aqal2iwow5oh/include -Wall -O2 -g -DNDEBUG -fPIC -std=gnu++17 -MD -MT libmamba/CMakeFiles/libmamba.dir/src/core/channel.cpp.o -MF CMakeFiles/libmamba.dir/src/core/channel.cpp.o.d -o CMakeFiles/libmamba.dir/src/core/channel.cpp.o -c /tmp/sam/spack-stage/spack-stage-micromamba-1.0.0-hsyfletrgwe3rpkmoisedxw2zts5r7it/spack-src/libmamba/src/core/channel.cpp
/tmp/sam/spack-stage/spack-stage-micromamba-1.0.0-hsyfletrgwe3rpkmoisedxw2zts5r7it/spack-src/libmamba/src/core/singletons.cpp:9:10: fatal error: curl/urlapi.h: No such file or directory
    9 | #include <curl/urlapi.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libmamba/CMakeFiles/libmamba.dir/build.make:141: libmamba/CMakeFiles/libmamba.dir/src/core/singletons.cpp.o] Error 1
```

However, `curl` is included as a build-dependency, and it is `find_package`d by CMake ([this line](https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L425),

File: var/spack/repos/builtin/packages/libreproc/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libreproc/package.py]: "run" doesn't make sense for zlib since there is no `prefix.bin` directory

File: var/spack/repos/builtin/packages/libsolv/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libsolv/package.py]: You don't need to define both the true and false cases, I think the opposite will be obvious

File: var/spack/repos/builtin/packages/tl-expected/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tl-expected/package.py]: Commit hashes aren't good version names because they don't sort properly. Instead, I would recommend a version like `2022-11-24` (the date of the commit)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/tl-expected/package.py]: Also, how does this fetch properly? Don't you need to specify the `git` attribute and download from git?

File: var/spack/repos/builtin/packages/micromamba/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/micromamba/package.py:154]: @tldahlgren what's the easiest way to support this test both at build time and post-install?

File: var/spack/repos/builtin/packages/micromamba/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/micromamba/package.py]: Run probably not needed

File: var/spack/repos/builtin/packages/micromamba/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/micromamba/package.py]: run probably not needed

File: var/spack/repos/builtin/packages/tl-expected/package.py
Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/tl-expected/package.py]: It uses the <https://github.com/TartanLlama/expected/archive/b74fecd.tar.gz> which GitHub resolves properly.

Is it preferred to talk to GitHub repos over HTTP (archive tarball) or Git?

Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/tl-expected/package.py]: ... which is why the version is named after the commit hash.

File: var/spack/repos/builtin/packages/micromamba/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/micromamba/package.py]: `tty.warn` is better than `warnings` since that's how all Spack error/warning/debug/info messages are printed. But I wouldn't do this here, I would just add a conflict or even remove the option until we get it working.

File: var/spack/repos/builtin/packages/tl-expected/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/tl-expected/package.py]: Gotcha. We generally prefer HTTP over Git because it can be checksummed, but git hashes also work for this purpose. I think most package specify the commit and download via git instead of using HTTP for a random commit. This also makes the version more meaningful.

File: var/spack/repos/builtin/packages/tl-expected/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tl-expected/package.py:22]: since 2022 > 1, this version will be considered newer and will install by default. Is this what you want? Maybe we should put it above 1.0.0 just to clarify this.

**Review by adamjstewart — DISMISSED** [review]
LGTM, not sure if anyone else would be interested in reviewing this.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
